datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -262,6 +262,65 @@ fn date_trunc_single(granularity: &str, value: i64) -> Result<i64> {
     Ok(value.unwrap().timestamp_nanos())
 }
 
+fn _date_trunc(
+    tu: TimeUnit,
+    value: &Option<i64>,
+    granularity: &str,
+    f: impl Fn(Option<i64>) -> Result<Option<i64>>,
+    tz_opt: &Option<Arc<str>>,
+) -> Result<ColumnarValue, DataFusionError> {
+    let scale = match tu {
+        TimeUnit::Second => 1_000_000_000,
+        TimeUnit::Millisecond => 1_000_000,
+        TimeUnit::Microsecond => 1_000,
+        TimeUnit::Nanosecond => 1,
+    };
+    let nano = (f)(Some(value.unwrap() * scale))?;
+    let result = match tu {
+        TimeUnit::Second => {
+            let value = match granularity {
+                "minute" => Some(nano.unwrap() / 1_000_000_000),
+                "second" => Some(nano.unwrap() / 1_000_000_000),
+                "millisecond" => Some(nano.unwrap() / 1_000_000),
+                "microsecond" => Some(nano.unwrap()),
+                _ => Some(nano.unwrap() / 1_000_000_000),
+            };
+            ScalarValue::TimestampSecond(value, tz_opt.clone())
```

Comment:
```suggestion
            let value = Some(nano.unwrap() / 1_000_000_000);
            ScalarValue::TimestampSecond(value, tz_opt.clone())
```

As output unit is second, smaller granularity can be truncated simply.